build: improve error message for upgrade test helper

devversion opened the pull request:
Currently whenever the upgrade test helper fails to load a given AngularJS version, the error that will be rejected is technically not an error because the `onerror` callback is not returning an error, but an "ErrorEvent".

Since that `ErrorEvent` is basically just rejected, browsers will print the error as followed:

```
Failed: [object Event]
```

This is not helpful at all and also implies that there _might_ be more information hidden within the `Event` instance. Unfortunately that's not the case (at least on browsers we test against) and the logic to extract the data from the event would be not worth the effort, we just return a simple custom `Error` that won't imply that there is more information hidden.

mary-poppins commented:
You can preview 477f300 at https://pr27706-477f300.ngbuilds.io/.

mary-poppins commented:
You can preview 7893a3e at https://pr27706-7893a3e.ngbuilds.io/.

gkalpak commented on packages/upgrade/test/common/test_helpers.ts:
```diff
@@ -69,7 +69,18 @@ export function createWithEachNg1VersionFn(setNg1: typeof setAngularJSGlobal) {
                                             script.async = true;
-                                            script.onerror = reject;
+                                            script.onerror =
+                                                (message, url, line, column, error) => {
```
AFAICT, these arguments only apply to `window.onerror`. `script.onerror` is passed a error `Event` is fired. From [MDN][1]:

> - When a **JavaScript runtime error** (including syntax errors and exceptions thrown within handlers) occurs, an error event using interface `ErrorEvent` is fired at `window` and `window.onerror()` is invoked. [...]
> - When a resource (such as an `<img>` or `<script>`) fails to load, an error event using interface `Event` is fired at the element that initiated the load, and the `onerror()` handler on the element is invoked. [...]
> [...]
> For historical reasons, different arguments are passed to `window.onerror` and `element.onerror` handlers (as well as on error-type `window.addEventListener` handlers).

Did you see something different happening?

[1]: https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror

devversion committed:
build: improve error message for upgrade test helper

Currently whenever the upgrade test helper fails to load a given AngularJS version, the error that will be rejected is technically not an error because the `onerror` callback is not returning an error, but an "ErrorEvent".

Since that `ErrorEvent` is basically just rejected, browsers will print
the error as followed:

```
Failed: [object Event]
```

This is not helpful at all and also implies that there _might_ be more
information hidden within the `Event` instance. Unfortunately that's not
the case (at least on browsers we test against) and the logic to extract
the data from the event would be not worth the effort, we just return a
simple custom `Error` that won't imply that there is more information
hidden.

mary-poppins commented:
You can preview 00543b9 at https://pr27706-00543b9.ngbuilds.io/.
